notebooks/problem-sets/qiskit_programming_australia.ipynb

```diff
@@ -0,0 +1,490 @@
rCrBI8n4cMkhbmu7aJP8wSFmIBq33VOY/o9fXxML5+e0laPbPt4Rts/npcHHtB+Xl0V5T/KVPedUPxOdo65SqD9Ri2PEWhnacxePqykF1ORFf2x8SiNg69Td2sMtH+bvilpMmYTm77iD/LMWKS9yB79xpMfxuM6hVUCdybRMfpdPF+ILxpj1h3tQrjOtSJOki1jq76Y/kxZ5e+yDK63gXup01IHsUdtPR7b4xL9i05irLC5AI4WZB+nDXLPEjnyXDKh7gfrb4C77BBto51Xavxj0zIRuWTatAChGROJGUPtQO8299s3F/gdGcsYTlH8biZWwht44nH0cWA7OQD1BJxF4AbsD8voWCKTclNeO6Y9pElLOQe0Y79YInPb7bf7/bt2vs3v9ebABrh39A/ChIFnqcyxeltIlnfgYWXQl4CzLYgdxXmmjcJkzeTrAu7E/MbrwZp3wLs9uH2BRK2QI4xddfa9ljJoR7stuRagx+/ewF0EZPtBHgfi4SU/baat6+qxbvMNmnmD/7HHHvOYuOeee/y05FL2VNGfWV8JqK20Q8zOo3oR9eV3td3ja14ic1Yg3l9CVfvR+oyTgHdiENB5lmAq3rlfIFg7xuy+TWvZb76m2nm9dk66gu0e6ycSNdghWxN0zcXk3oQYYxMD1ZzX+QFSZbn0Vzu0gTt9N5O8tQtc0u9lAnf0Gf2O3iEsDbiL/x4rxPVBfe6X4p4ujMBEcc09V1wnDZ8KSp82PavrTSjzc17f83ipx43HjDJk7CRujx3nUTrAE4Ad8B49fav0NE8LL6UAL/xeSyHvgETAesY/66jT7kuibepTh7qfZ6HZ41IZ4IvHyZQyMpZmqaOdNrydVJYnneoX9JRxhyZY2EyWar2il6MZh4xV3qvii935LgwyPDbUbUv95CfwGxuOHQa4b76YOpaDa31jA9w7uieCnf3cm20hZfDjpUq2SQJZpkEomUXjdX9Hnhy+rooQITzX1buUIEyeGEiZTguuFwU0LeGU+26r0nS1w/dUx7zAPYooS2YoB/6xkwHeAhRIo1im0Lkdt9N+gPujjz5qHt13330G7tC9zrRP4hfGAUBy6oSehBw7WZ3Vevadr+ldhBff0cuo2maMDyuh+IvBlVODdwA7a9a1BGbntXpp+xYtj7n7pmrXrdo1Ri+hskd7xlPy6YJJan7XBJagm0sNT+Fv0tS8LnnudEmTWOlTb1lWgDvt5ugL3BtaajpmjWbNPzdwr3kb+to8hidMog3cFRMC3MzTZKzjdv7Wbf8s+Uzflnm4t+UgR31tdKrURR6nTRpiBRkml+lydd4E9XUZ+JU0xJTlNrXGTNe1spzyHMcKu4EB4NHJLE0ALLF93qUUGPf0uQ/Ju8e/rgHcGwfMZQTc6buMkYzPaf1JukVCdECfckhDH9leCqwD3I9omSbjkHJ4mfTmm282XujzxHCLpNRyQznIPts/4r1nO8hljWvaQRt4xw/cQ0COcFRSN3RMCtHb5H9JS9bAT+Qh/6161+xSm0hPauus9zbAfQzHECLWnGVbSAwJe7qj8DD8CY1YTxDwAFi8nAgh3mcGNevdmU1T5kF2K9GA7L1bieqLEgots8a00ceYjLmfbeT43Sh88SBKaUz2UdnKQ7sROtqMYqIMr3WXEK7zo1raS9+9XnvcaecD999v4D6uzZfC9Rj0k+/oUezLb3m7x+rwO1X1mpaFHdUa6FMCd83A3toiXwa0ax92f1RJu8bwJVS2fcTzXu3ZOfLMZ2wRU0R++3RUOPx1yL2k5WLO6zQZa4lHGYu/MUxbLo0mV9TDRIWxx3npcRxbnsop75XnVNH+XVTbnJZpyvMmQetkFuDe8I4yah6luJqr+Tny5JXAHePNUpkxMlzm31KPq7rQd76nuom5mnz+DU31dQjZUg73ylD2nc4BHAYdud4Vk65Om6KoA++l+VHfz72S/+V57iemDD649Ix2kUL/s5YYsHFl7WhIurWP1Q7GO+Me4I7M+9B12o8DBTtmR5F+XyohI2deij021VjHjJWegfSzBNJ7fNa87ZOffoqnHV0APkBG+VK5lzdp4sjuRthOvO0lDmnTNm6Mc53JaLN/uyalLAVbRog+++53v1t961vfchWf+MQnqh/5kR+pPvaxj3kSMqneAP/vfe971T/8wz/4WyosEfr4xz9e/cIv/EJ17733Tsr+gb63Ae4Tuhelx7aQeN7ZFnKXBr0fMUuQEhpxniDYKEzWeDeej1qJ8uLmG6/rJUCVDWDHs8NMNN7oSYJp4yQirIBCzAyx6RbNzg/tHco796APIaJOT1xqpT9OObTJoN1MWFBKrHdnqRHt9Q4zAvEAqXUM9BvGD4/7E3o5lfbeX3vc+yjidWwT6OrsSb0I9Z4mkK+9Ux09/IbXtFcvvlvtePek1rprKDSDutUCxohA+84rNXllj/YPaYkXH1ZSvEugfcfeXc14ythJTEk+r2PGUjMGc68ej1vuMTbHhPRBk6KVlv7iiPwZuKtPkassFekquhzXOU/clT7XutLkWuKk7YqZID788MPm0wMPPDBxgpi2w6t2aO7RXwq037qn1kHQ0nw3opU5fcLli0se9aHv1fW6Lp2Ttkzf0EDiImwBXDV90MP18Ci/0UlbrhXpc68sjzppq+uu0xZVTz2NvPMyOsCdScAl+U4L/aGDpyw4TfwFTrU+/IF3LJFBDjjWVf92dVjGWNnvXenGXYMHzTEuUcd18swbGMOT8kc+sZE8ieeJD+9XcD1YgD6zp1wYgXfibIcLHNKmzTLUUS/Xsb0HNSHlaT8gnkncMsLL2knvO9/5TvW1r32t+sd//EdjiM985jMVx8/+7M9Whw4dmlht3i3793//9+qv/uqvqteFlQDtn/70p6vPf/7zFTrycg0b4D6h5zMDZneZszpQFm1PVQQyYt2lUErPB+cID9svUX5eguKREgGvex4lXcFMWGnHhloJcT/1j01b3CCt6SZ/cb19ShqONnD3ciEpD
```

Comment:
@Astri-Cornish FYI, `to_numpy_matrix` has been removed in networkx 3: 

see https://networkx.org/documentation/stable/release/migration_guide_from_2.x_to_3.0.html#replacing-numpy-scipy-matrices-with-arrays

got this error when testing:
<img width="709" alt="image" src="https://user-images.githubusercontent.com/13156555/223346440-8ad61910-4d15-45b6-b5d6-8350f0d114d2.png">

Reply:
Thanks for this! I have just added a commit to fix this here and in the grader